libra-node/src/main_node.rs

```diff
@@ -298,6 +300,18 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
             node_config.base.waypoint.expect("No waypoint in config"),
         );
 
+        // Set up to listen for network configuration changes from StateSync.
+        if let Some(conn_mgr_reqs_tx) = network_builder.conn_mgr_reqs_tx() {
```

Comment:
This returns an option, we should use an expect to enforce that this is some or panic otherwise:
https://doc.rust-lang.org/std/option/enum.Option.html#method.expect

Reply:
Why should we panic here?  we only have a connectivity_manager if _enable\_remote\_authentication is enabled (main_node.rs:146), which is not guaranteed to be the case.

Reply:
we cannot form a network otherwise :)

let's just panic for now and we can deal with the plumbing issues later.

of course one could argue we have bigger issues if someone runs this without auth...

I'll leave it up to you.

Reply:
In the essence of minimal changes, I'll defer removing the `enable_remote_authentication` to a follow up.  
1). it's feature creep
2). I lack the context to make that judgement call/appreciate implications thereof
3). 9 hour round trips exacerbate (2)

Reply:
Will defer removing enable_remote_authentication for a couple of reasons:

1). scope creep
2). I'm missing context to make judgement calls
3). timezones will make it inefficient for me to do that removal